dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -1305,10 +1274,139 @@ private Response response ( String response, Boolean error ) {
      */
     private Response response ( String response, Boolean error, String contentType ) {
         if ( error ) {
-            return Response.status( HttpStatus.SC_INTERNAL_SERVER_ERROR ).entity( response ).build();
+            return response(response, HttpStatus.SC_INTERNAL_SERVER_ERROR);
         } else {
             return Response.ok( response, contentType ).build();
         }
     }
 
+    private Response response ( String response, int status ) {
+        return Response.status( status ).entity( response ).build();
+    }
+
+    private class IntegrityDataRequestChecker implements Runnable{
+
+        private final User loggedUser;
+        private final HttpSession session;
+        private final PublishingEndPoint endpoint;
+        private final String integrityDataRequestID;
+
+        public IntegrityDataRequestChecker(
+                final User loggedUser,
+                final HttpSession session,
+                final PublishingEndPoint endpoint,
+                final String integrityDataRequestID) {
+
+            this.loggedUser = loggedUser;
+            this.session = session;
+            this.endpoint = endpoint;
+            this.integrityDataRequestID = integrityDataRequestID;
+        }
+
+        @CloseDBIfOpened
+        public void run(){
+
+            boolean processing = true;
+
+            while(processing) {
+
+                Response response = null;
+
+                response = statusIntegrityCheckerRequest();
+
+                if (response.getStatus() == HttpStatus.SC_OK) {
+
+                    processing = false;
+
+                    InputStream zipFile = response.readEntity(InputStream.class);
```

Comment:
Codacy found an issue: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)